tasks/node_update.rb

```diff
@@ -6,19 +6,19 @@
 require 'puppet'
 
 def node_update(availability, role, label_add, label_rm, node)
-  cmd_string = 'docker node update'
-  cmd_string += " --availability #{availability}" unless availability.nil?
-  cmd_string += " --role #{role}" unless role.nil?
+  cmd_string = ['docker', 'node', 'update']
+  cmd_string += [" --availability #{availability}"] unless availability.nil?
+  cmd_string += [" --role #{role}"] unless role.nil?
 
   if label_add.is_a? Array
     label_add.each do |param|
-      cmd_string += " --label-add #{param}"
+      cmd_string += [" --label-add #{param}"]
```

Comment:
See above comment

Reply:
See above.

Reply:
In this case you have changed from concatenating a string to an array so we should generally lean towards the more idiomatic approach.